src/MergePlugin.php

```diff
@@ -135,6 +144,21 @@ public static function getSubscribedEvents()
     }
 
     /**
+     * Handle an event callback for initialization.
+     *
+     * @param \Composer\EventDispatcher\Event $event
+     */
+    public function onInit(\Composer\EventDispatcher\Event $event)
+    {
+        $this->state->loadSettings();
+        // @todo devMode cannot be properly detected at this stage, but
+        //       merging it should not hurt.
+        $this->state->setDevMode(true);
```

Comment:
I think the way to fix this issue would be to split the dev mode processing in `ExtraPackage::mergeInto` into a separate function. Then `MergePlugin::mergeFile` could be adjusted to call or not call the new function as appropriate. The guard against double processing a file there would then need to be adjusted to keep track of both types of merges. This would a bit of new complexity, but it shouldn't be too horrible.